toolbox/errorfinder/main.go

```diff
@@ -0,0 +1,518 @@
+
+				// Delete date information that are possibly present at the beginning of each line.
+				line = strings.TrimLeft(line, "\\d-T:.Z	info")
+				lineSplit := strings.Split(line, " ")
+				newLineSplit := []string{}
+
+				for indexD, section := range lineSplit {
+
+					// Ignore empty strings
+					if strings.Compare(section, "") == 0 {
+						continue
+					}
+
+					// If word does not contain key words, generalize word to characters necessary.
+					if !strings.Contains(strings.ToLower(section), "error") &&
+						!strings.Contains(strings.ToLower(section), "fail") && !strings.Contains(strings.ToLower(section), "warn") {
+						if indexD > 0 && strings.Compare(lineSplit[indexD-1], "container") == 0 {
+							newLineSplit = append(newLineSplit, "\\s")
+						} else if strings.Contains(section, "\\d") {
+							newLineSplit = append(newLineSplit, "\\s")
+						} else if strings.Contains(section, "/") {
+							newLineSplit = append(newLineSplit, "\\s")
+						} else {
+							newLineSplit = append(newLineSplit, section)
+						}
+					} else {
+
+						// If word contains keyword, append it to list.
+						newLineSplit = append(newLineSplit, section)
+					}
+				}
+
+				// Get general error message not the detailed eplanation that usually comes after colon.
+				line = strings.Join(newLineSplit, " ")
+				if strings.Contains(line, ":") {
+					indexColon := strings.Index(line, ":")
+					if indexColon < 20 {
+						newLine := strings.Replace(line, ":", "<", 1)
+						if strings.Contains(newLine, ":") {
+							indColon := strings.Index(newLine, ":")
+							if strings.Contains(line[:indColon], "error") || strings.Contains(line[:indColon], "warn") || strings.Contains(line[:indColon], "fail") {
+								line = line[:indColon]
+							}
+
+						}
+					} else {
+						if strings.Contains(line[:indexColon], "error") || strings.Contains(line[:indexColon], "warn") || strings.Contains(line[:indexColon], "fail") {
+							line = line[:indexColon]
+						}
+					}
+				}
+
+				// If error line contains keyword "warn", it is a warning and should be placed in warning map.
+				if strings.Contains(strings.ToLower(line), "warn") {
+					var warningPrs []string
+					if warningMap[line] == nil {
+						warningPrs = []string{filePath}
+					} else {
+						warningPrs = warningMap[line]
+						if !contains(warningPrs, filePath) {
+							warningPrs = append(warningPrs, filePath)
+						}
+					}
+					warningMap[line] = warningPrs
+					// Else it's an error and should be placed to error map.
+				} else {
+					var prSlice []string
+					if errorMap[line] == nil {
+						prSlice = []string{filePath}
+					} else {
+						prSlice = errorMap[line]
+						if !contains(prSlice, filePath) {
+							prSlice = append(prSlice, filePath)
+						}
+					}
+					errorMap[line] = prSlice
+				}
+			}
+		}
+	}
+	return errorMap, warningMap, nil
+}
+
+// Read destination csv file to get previously generated errors and warnings.
```

Comment:
What is your csv format? 
Is your format: error type (error or warning),  error message, list of paths

Reply:
I have two lines in my csv file. One containing only the word "error" and one containing only "warning". The lines under "error" line are errors. Each line except the "error" and "warning" lines are composed of error/warning message, number of paths that contain the error/warning and the list of paths.